tests/unit-core/test-container-operators.c

```diff
@@ -0,0 +1,118 @@
+  TEST_ASSERT (!jerry_value_is_error (value));
+  jerry_release_value (value);
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_GET, map, &key_str, 1);
+  TEST_ASSERT (jerry_get_number_value (value) == 10);
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_HAS, map, &key_str, 1);
+  TEST_ASSERT (jerry_get_boolean_value (value));
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_SIZE, map, NULL, 0);
+  TEST_ASSERT (jerry_get_number_value (value) == 1);
+
+  key_str = jerry_create_string ((jerry_char_t *) "number2");
+  number = jerry_create_number (11);
+  jerry_value_t args2[2] = {key_str, number};
+  value = jerry_container_operation (JERRY_CONTAINER_OP_SET, map, args2, 2);
+  jerry_release_value (value);
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_SIZE, map, NULL, 0);
+  TEST_ASSERT (jerry_get_number_value (value) == 2);
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_DELETE, map, &key_str, 0);
+  jerry_release_value (value);
+  TEST_ASSERT (jerry_get_boolean_value (value));
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_SIZE, map, NULL, 0);
+  TEST_ASSERT (jerry_get_number_value (value) == 1);
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_CLEAR, map, NULL, 0);
+  TEST_ASSERT (jerry_value_is_undefined (value));
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_SIZE, map, NULL, 0);
+  TEST_ASSERT (jerry_get_number_value (value) == 0);
+
+  jerry_release_value (map);
+  jerry_release_value (value);
+
+  number = jerry_create_number (10);
+  jerry_value_t set = jerry_create_container (JERRY_CONTAINER_TYPE_SET, NULL, 0);
+  TEST_ASSERT (jerry_get_container_type (set) == JERRY_CONTAINER_TYPE_SET);
+  value = jerry_container_operation (JERRY_CONTAINER_OP_SET, set, &number, 1);
+  TEST_ASSERT (!jerry_value_is_error (value));
+  jerry_release_value (value);
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_HAS, set, &number, 1);
+  TEST_ASSERT (jerry_get_boolean_value (value));
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_SIZE, set, NULL, 0);
+  TEST_ASSERT (jerry_get_number_value (value) == 1);
+
+  number = jerry_create_number (11);
+  value = jerry_container_operation (JERRY_CONTAINER_OP_SET, set, &number, 1);
+  jerry_release_value (value);
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_SIZE, set, NULL, 0);
+  TEST_ASSERT (jerry_get_number_value (value) == 2);
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_DELETE, set, &number, 0);
+  jerry_release_value (value);
+  TEST_ASSERT (jerry_get_boolean_value (value));
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_SIZE, set, NULL, 0);
+  TEST_ASSERT (jerry_get_number_value (value) == 1);
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_CLEAR, set, NULL, 0);
+  TEST_ASSERT (jerry_value_is_undefined (value));
+
+  value = jerry_container_operation (JERRY_CONTAINER_OP_SIZE, set, NULL, 0);
+  TEST_ASSERT (jerry_get_number_value (value) == 0);
+
```

Comment:
Missing tests:
* providing a non-container type as the input argument
* providing a container type as input object, but incorrectly specifying the argument numbers.
* returning a non trivial type (eg: objects) to test if the returned object if correctly refcounted (must call the `jerry_release_value` )